### PR TITLE
Add Prettier for formatting and format all files

### LINKED ISSUE #1457: Use Prettier for code formatting and upgrade Dusk ESlint config to 3.0.2
#### Summary

Prettier needs be used for code formatting in the web wallet module, in combination with the new ESlint configuration (3.0.0).  We have created a [shared configuration for Prettier](https://github.com/dusk-network/prettier-config).  Note that the README explains how to use this in a (Node) project.

### PR DESCRIPTION
**Pull request by kieranhall**
Resolves #1457

### CODE REVIEW

File: web-wallet/src/lib/components/GasControls/GasControls.svelte [hunk 1]
```diff
@@ -1,95 +1,91 @@
...
-	<Textbox
-		bind:value={price}
-		className="gas-control__input"
-		max={limit}
-		min={priceLower}
-		on:input={checkGasLimits}
-		required
-		type="number"
-	/>
+  <span class="gas-control__label"> Price: (lux) </span>
```
Comment (nortonandreev) [inline web-wallet/src/lib/components/GasControls/GasControls.svelte:49]: Seems like there's an additional whitespace here.

File: web-wallet/src/lib/components/GasControls/GasControls.svelte [hunk 2]
```diff
@@ -1,95 +1,91 @@
...
-	<Textbox
-		bind:value={limit}
-		className="gas-control__input"
-		max={limitUpper}
-		min={limitLower}
-		on:input={checkGasLimits}
-		required
-		type="number"
-	/>
+  <span class="gas-control__label"> Gas Limit: (unit) </span>
```
Comment (nortonandreev) [inline web-wallet/src/lib/components/GasControls/GasControls.svelte:62]: Again – the whitespace.

File: web-wallet/src/lib/components/__tests__/OperationResult.spec.js [hunk 3]
```diff
@@ -1,117 +1,127 @@
...
 describe("OperationResult", () => {
-	const delay = 1000;
+  const delay = 1000;
 
-	/** @type {(delay: number) => Promise<any>} */
-	const rejectAfter = ms => new Promise((resolve, reject) => {
-		setTimeout(() => reject(new Error("some error")), ms);
-	});
+  /** @type {(delay: number) => Promise<any>} */
+  const rejectAfter = (ms) =>
```
Comment (nortonandreev) [inline web-wallet/src/lib/components/__tests__/OperationResult.spec.js:12]: Not related to the PR, but still...

There's a mismatch between the function param and the doc comment above (`delay` vs `ms`).

File: web-wallet/src/lib/components/__tests__/OperationResult.spec.js [hunk 4]
```diff
@@ -1,117 +1,127 @@
...
+    new Promise((resolve, reject) => {
+      setTimeout(() => reject(new Error("some error")), ms);
+    });
 
-	/** @type {(delay: number) => Promise<any>} */
-	const resolveAfter = ms => new Promise(resolve => { setTimeout(resolve, ms); });
+  /** @type {(delay: number) => Promise<any>} */
+  const resolveAfter = (ms) =>
```
Comment (nortonandreev) [inline web-wallet/src/lib/components/__tests__/OperationResult.spec.js:18]: Same as above.

File: web-wallet/src/lib/dusk/components/ErrorDetails/ErrorDetails.svelte [hunk 5]
```diff
@@ -1,31 +1,31 @@
...
-		<summary class="dusk-error-details__summary">
-			{summary}
-		</summary>
-		<pre class="dusk-error-details__error">
+  <details class={classes}>
+    <summary class="dusk-error-details__summary">
+      {summary}
+    </summary>
+    <pre class="dusk-error-details__error">
 			{error.message}
```
Comment (nortonandreev) [inline web-wallet/src/lib/dusk/components/ErrorDetails/ErrorDetails.svelte:28]: This is an odd issue that @deuch13 had experienced in his PR as well. Looking at the file, it looks fine

Screenshot 2024-02-28 at 11 47 09

however, if you remove the white space and press enter, you will see GitHub showing the change in the file, as follows:

Screenshot 2024-02-28 at 11 46 09

Which resolves the issue.


Seems like it's a character that is encoded differently (and rendered differently in different environments) – for example, if you try to copy and paste the whitespace in a text field, you will see there is a difference, as shown in the video below:

https://github.com/dusk-network/rusk/assets/34094428/3991deee-2193-46b1-bc76-66448350f8ad

It looks like it is a bug with the linter... doesn't happen each time, but this is the second time we are seeing this.

File: web-wallet/src/routes/(app)/settings/__tests__/__snapshots__/page.spec.js.snap [hunk 6]
```diff
@@ -161,8 +161,8 @@ exports[`Settings > Resetting the wallet > should show an error if clearing loca
           <p
             class="s-EBshooXNO1eQ"
           >
-            These functionalities are currently disabled in the beta
-					version of the web wallet and will be enabled in a future update.
+            These functionalities are currently disabled in the beta version of
```
Comment (nortonandreev) [inline web-wallet/src/routes/(app)/settings/__tests__/__snapshots__/page.spec.js.snap:164]: I will create an issue to extract this as a constant that we can reuse here – as we have this message in 3 places.

File: web-wallet/src/lib/components/__tests__/OperationResult.spec.js (hunk 3 above)
Comment (ascartabelli) [reply web-wallet/src/lib/components/__tests__/OperationResult.spec.js:12]: > Not related to the PR, but still...
> 
> There's a mismatch between the function param and the doc comment above (`delay` vs `ms`).

Doesn't the type checker usually show an error for this?

File: web-wallet/src/lib/dusk/components/ErrorDetails/ErrorDetails.svelte (hunk 5 above)
Comment (ascartabelli) [reply web-wallet/src/lib/dusk/components/ErrorDetails/ErrorDetails.svelte:28]: > It looks like it is a bug with the linter... doesn't happen each time, but this is the second time we are seeing this.

Was the other time also about white spaces inside a tag?

(this can be the reason: how to interpret it and whether such white space should be left alone or not)

Comment (deuch13) [reply web-wallet/src/lib/dusk/components/ErrorDetails/ErrorDetails.svelte:28]: No, it was white space in a variable definition. `let   variable`

**Comment by nortonandreev** [comment]
Screenshot 2024-02-28 at 13 00 38

The animation on the home page is not appearing on this branch.

File: web-wallet/src/routes/(welcome)/setup/Animation.svelte
Comment (nortonandreev) [inline web-wallet/src/routes/(welcome)/setup/Animation.svelte:1]: As posted on the main PR's discussion, the whitespaces added after the custom shorthand `@p` breaks the code.

File: web-wallet/src/routes/(welcome)/setup/PasswordSetup.svelte [hunk 7]
```diff
@@ -1,86 +1,86 @@
...
+        placeholder="Confirm Password"
+      />
+      {#if password.length < 8}
+        <span class="confirm-password--meta"
+          >Password must be at least 8 characters</span
+        >
+      {:else if confirmPassword && password !== confirmPassword}
+        <span
+          class="confirm-password--meta
+						confirm-password--meta--error">Passwords do not match</span
```
Comment (nortonandreev) [inline web-wallet/src/routes/(welcome)/setup/PasswordSetup.svelte:52]: Again the problem with the awkward whitespace.

File: web-wallet/src/routes/(welcome)/setup/restore/MnemonicAuthenticate.svelte [hunk 8]
```diff
@@ -1,28 +1,35 @@
...
-	$: {
-		if (enteredMnemonicPhrase.filter(word => word !== "").length === wordLimit && !isValid) {
-			toast("error", "Invalid mnemonic phrase", mdiAlertOutline);
-		}
-	}
+  $: isValid = validateMnemonic(
+    enteredMnemonicPhrase.join(" "),
+    wordlists.english
+  );
+  $: {
```
Comment (nortonandreev) [inline web-wallet/src/routes/(welcome)/setup/restore/MnemonicAuthenticate.svelte:20]: I will open an issue for this one – the additional code block is not needed. `$: if...` should be enough.

File: web-wallet/.eslintrc.cjs [hunk 9]
```diff
@@ -1,73 +1,69 @@
...
+    node: true,
+  },
+  extends: [
+    "@dusk-network/eslint-config/js",
+    "@dusk-network/eslint-config/svelte",
+  ],
+  globals: {
+    CONFIG: false,
+  },
+  overrides: [
```
Comment (nortonandreev) [inline web-wallet/.eslintrc.cjs:14]: Do we need the overrides?

File: web-wallet/src/lib/components/__tests__/OperationResult.spec.js (hunk 3 above)
Comment (nortonandreev) [reply web-wallet/src/lib/components/__tests__/OperationResult.spec.js:12]: Opened issue https://github.com/dusk-network/rusk/issues/1471.

File: web-wallet/src/lamb.d.ts [hunk 10]
```diff
@@ -1,687 +1,683 @@
...
+  /* ------------------------- *
+   * ***** UTILITY TYPES ***** *
+   * ------------------------- */
+
+  type AnyFunction = (...args: any) => any;
+
+  type FoldAccumulatorCallback<
+    L extends ArrayLike<any>,
+    R,
+    I extends keyof L & number = any,
```
Comment (deuch13) [inline web-wallet/src/lamb.d.ts:11]: extra comma added to all type definitions

**Review by nortonandreev — COMMENTED** [review]
Thanks for the changes, @kieranhall! 💥

I have left some minor comments. The animation on the home page is broken which is the biggest issue I am seeing with the changes.

`web-wallet:` prefix is missing from the commit message and the change log hasn't been updated to reflect the changes delivered.

I have also opened 3 issues, that I uncovered while reviewing this PR: #1469, #1470, and #1471.

Keep in mind I haven't looked into the `.eslintrc.cjs` and the `.prettierrc.js` files too carefully, as I am by no means an expert in this type of configuration. I am sure @ascartabelli can be of much better help there.

File: web-wallet/src/lib/components/ContractOperations/ContractOperations.svelte [hunk 11]
```diff
@@ -1,67 +1,70 @@
...
-		mdiDatabaseOutline,
-		mdiHelpCircleOutline
-	} from "@mdi/js";
+  import { createEventDispatcher } from "svelte";
+  import {
+    mdiArrowDownBoldBoxOutline,
+    mdiArrowUpBoldBoxOutline,
+    mdiDatabaseArrowDownOutline,
+    mdiDatabaseOutline,
+    mdiHelpCircleOutline,
```
Comment (deuch13) [inline web-wallet/src/lib/components/ContractOperations/ContractOperations.svelte:10]: similar issue here there is an extra comma that shouldn't be added

File: web-wallet/src/lib/components/GasControls/GasControls.svelte (hunk 1 above)
Comment (deuch13) [reply web-wallet/src/lib/components/GasControls/GasControls.svelte:49]: Where exactly? I do not see it.

File: web-wallet/src/lib/components/OperationResult/OperationResult.svelte [hunk 12]
```diff
@@ -1,89 +1,80 @@
...
-		Icon,
-		Throbber
-	} from "$lib/dusk/components";
-	import { mdiCheckDecagramOutline, mdiCloseThick } from "@mdi/js";
-	import { makeClassName } from "$lib/dusk/string";
+  import {
+    AnchorButton,
+    ErrorDetails,
+    Icon,
+    Throbber,
```
Comment (deuch13) [inline web-wallet/src/lib/components/OperationResult/OperationResult.svelte:8]: Again this extra comma, do not know how it is added as it does not happen constantly as you can see 2 lines below.

File: web-wallet/src/lib/components/ContractOperations/ContractOperations.svelte (hunk 11 above)
Comment (ascartabelli) [reply web-wallet/src/lib/components/ContractOperations/ContractOperations.svelte:10]: Sadly this is how we decided to set up prettier.

File: web-wallet/src/lib/components/Send/Send.svelte [hunk 13]
```diff
@@ -1,337 +1,337 @@
...
+  import { validateAddress } from "$lib/dusk/string";
+  import { logo } from "$lib/dusk/icons";
+  import {
+    AnchorButton,
+    Badge,
+    Button,
+    Icon,
+    Textbox,
+    Wizard,
+    WizardStep,
```
Comment (deuch13) [inline web-wallet/src/lib/components/Send/Send.svelte:19]: Looks like this extra comma is added if there are more than 2 or 3 items in between the curly brackets

File: web-wallet/src/lib/components/Send/Send.svelte [hunk 14]
```diff
@@ -1,337 +1,337 @@
...
-			</OperationResult>
-		</WizardStep>
-	</Wizard>
+  <Wizard steps={4} let:key>
+    <WizardStep
+      step={0}
+      {key}
+      backButton={{
+        action: resetOperation,
+        disabled: false,
```
Comment (deuch13) [inline web-wallet/src/lib/components/Send/Send.svelte:94]: I've also noticed it always adds a comma after the last property of an object

File: web-wallet/src/lib/components/Send/Send.svelte [hunk 15]
```diff
@@ -1,337 +1,337 @@
...
+            variant="secondary"
+            on:click={() => {
+              scanQrComponent.startScan();
+            }}
+            text="SCAN QR"
+          />
+        </div>
+        <Textbox
+          className="operation__send-address
+						{!addressValidationResult.isValid ? 'operation__send-address--invalid' : ''}"
```
Comment (deuch13) [inline web-wallet/src/lib/components/Send/Send.svelte:173]: shouldn't we still use `" "` instead of `' '`

File: web-wallet/src/lib/components/ContractOperations/ContractOperations.svelte (hunk 11 above)
Comment (deuch13) [reply web-wallet/src/lib/components/ContractOperations/ContractOperations.svelte:10]: I see, but it only adds that extra comma if there are more than 2 items

File: web-wallet/src/lib/components/__tests__/ContractOperations.spec.js [hunk 16]
```diff
@@ -1,95 +1,99 @@
...
-	expect,
-	it,
-	vi
-} from "vitest";
-import {
-	cleanup,
-	fireEvent,
-	render
-} from "@testing-library/svelte";
+import { afterEach, describe, expect, it, vi } from "vitest";
```
Comment (deuch13) [inline web-wallet/src/lib/components/__tests__/ContractOperations.spec.js:1]: According to what I've noticed there should be an extra comma after `vi`

File: web-wallet/src/lib/components/__tests__/ContractOperations.spec.js [hunk 17]
```diff
@@ -1,95 +1,99 @@
...
+import { cleanup, fireEvent, render } from "@testing-library/svelte";
```
Comment (deuch13) [inline web-wallet/src/lib/components/__tests__/ContractOperations.spec.js:2]: same here, there should be an extra comma after `render`

File: web-wallet/src/lib/containers/StakeContract/StakeContract.svelte [hunk 18]
```diff
@@ -1,171 +1,171 @@
...
-		Throbber
-	} from "$lib/dusk/components";
+  import { createCurrencyFormatter } from "$lib/dusk/currency";
+  import { getLastTransactionHash } from "$lib/transactions";
+  import { operationsStore, settingsStore, walletStore } from "$lib/stores";
+  import {
+    ContractOperations,
+    ContractStatusesList,
+    Stake,
+  } from "$lib/components";
```
Comment (deuch13) [inline web-wallet/src/lib/containers/StakeContract/StakeContract.svelte:23]: The rule that adds the extra comma seems off line 18 does not have the extra comma added but on line 22 it was added. This needs to be investigated further.

File: web-wallet/src/lib/components/GasControls/GasControls.svelte (hunk 1 above)
Comment (deuch13) [reply web-wallet/src/lib/components/GasControls/GasControls.svelte:49]: Ok, yes now i see it :)

File: web-wallet/src/lib/containers/StakeContract/StakeContract.svelte (hunk 18 above)
Comment (ascartabelli) [reply web-wallet/src/lib/containers/StakeContract/StakeContract.svelte:23]: > The rule that adds the extra comma seems off line 18 does not have the extra comma added but on line 22 it was added. This needs to be investigated further.

Seems the extra comma is added only in multi line imports. Try it [here](https://prettier.io/playground/).

(we are currently using the `es5` setting for that)

Comment (deuch13) [reply web-wallet/src/lib/containers/StakeContract/StakeContract.svelte:23]: This to me is very inconsistent and confusing. We either have a comma everywhere or we do not it should be simple.

Comment (kieranhall) [reply web-wallet/src/lib/containers/StakeContract/StakeContract.svelte]: > This, to me, is very inconsistent and confusing. We either have a comma everywhere or we do not it should be simple.

The comma-dangle rule has been deprecated from the ESlint library and we're following `es5` (as discussed with @ZER0) approach with ``. I believe the only options to changing this would be via the  `trailing-comma` setting, which is something we already discussed and decided on, or by adding [ESLint Stylistic](https://eslint.style/rules/js/comma-dangle); but I think it's a bit of an anti-pattern to go in that direction, as all the formatting should be done by Prettier now.  However, if it's not enough then I am open to discussing adding it.

File: web-wallet/.eslintrc.cjs (hunk 9 above)
Comment (kieranhall) [reply web-wallet/.eslintrc.cjs:14]: Yes, for working with Vitest, but [we will investigate](https://github.com/dusk-network/rusk/issues/1485) into removing `svelte/require-optimized-style-attribute` from here.

**Comment by nortonandreev** [comment]
> @nortonandreev I resolved a lot of your comments (inc. [this one](https://github.com/dusk-network/rusk/pull/1458#issuecomment-1968739389)). Can you give another look over the changes to verify the extra spaces are not added in other places?

Thanks! Sure, I will have a look tomorrow morning, if okay.

File: web-wallet/.editorconfig
Comment (ascartabelli) [inline web-wallet/.editorconfig:1]: Why did you remove the file altogether?

Comment (kieranhall) [reply web-wallet/.editorconfig:1]: After a discussion with @ZER0, it can't be moved to the root of the repo right now, so the preference is to just remove the file altogether. Apparently, it can be added back at a later date.

**Review by deuch13 — DISMISSED** [review]
All of my concerns have been explained to me, have nothing to object to.

**Review by ascartabelli — DISMISSED** [review]

**Comment by nortonandreev** [comment]
> @nortonandreev I resolved a lot of your comments (inc. [this one](https://github.com/dusk-network/rusk/pull/1458#issuecomment-1968739389)). Can you give another look over the changes to verify the extra spaces are not added in other places?

I will wait for the conflicts to be resolved and then will have one, final look, so to make sure something hasn't gone wrong during the merge.

File: web-wallet/.eslintrc.cjs (hunk 9 above)
Comment (kieranhall) [reply web-wallet/.eslintrc.cjs:14]: Also [we will investigate](https://github.com/dusk-network/eslint-config/issues/32) the offloading of our config to a community lead approach.

File: web-wallet/CHANGELOG.md [hunk 19]
```diff
@@ -142,9 +144,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 [#1469]: https://github.com/dusk-network/rusk/issues/1469
 [#1470]: https://github.com/dusk-network/rusk/issues/1470
 [#1471]: https://github.com/dusk-network/rusk/issues/1471
-[#1362]: https://github.com/dusk-network/rusk/issues/1362
```
Comment (nortonandreev) [inline web-wallet/CHANGELOG.md:145]: Problem with the conflict resolution – this shouldn't have been removed.

@kieranhall

**Review by nortonandreev — APPROVED** [review]
Thanks for the hard work, @kieranhall! 🔥 This refactor should significantly enhance our codebase, ensuring adherence to both community guidelines and organizational standards.

**Review by ascartabelli — APPROVED** [review]
